thingsboard_gateway/connectors/ftp/ftp_connector.py

```diff
@@ -0,0 +1,190 @@
+                           'MessagesSent': 0}
+        self.__log = log
+        self.__rpc_requests = []
+        self.__config = config
+        self.__connector_type = connector_type
+        self.__gateway = gateway
+        self.security = (self.__config["security"]["username"], self.__config["security"]["password"]) if \
+            self.__config["security"]["type"] == 'basic' else ('anonymous', 'anonymous@')
+        self.__tls_support = self.__config.get("TLSSupport", False)
+        self.setName(self.__config.get("name", "".join(choice(ascii_lowercase) for _ in range(5))))
+        self.daemon = True
+        self.__stopped = False
+        self.__requests_in_progress = []
+        self.__convert_queue = Queue(1000000)
+        self.__attribute_updates = []
+        self._connected = False
+        self.host = self.__config['host']
+        self.port = self.__config.get('port', 21)
+        self.__ftp = FTP_TLS if self.__tls_support else FTP
+        self.paths = [Path(path=obj['path'], with_sorting_files=True, poll_period=60, read_mode=obj['readMode'],
+                           max_size=obj['maxFileSize'], delimiter=obj['delimiter'], telemetry=obj['timeseries'],
+                           device_name=obj['devicePatternName'], device_type=obj.get('devicePatternType', 'Device'),
+                           attributes=obj['attributes'], txt_file_data_view=obj['txtFileDataView']) for obj
+                      in self.__config['paths']]
+
+    def open(self):
+        self.__stopped = False
+        self.start()
+
+    def run(self):
+        try:
+            with self.__ftp() as ftp:
+                self.__connect(ftp)
+
+                for path in self.paths:
+                    path.find_files(ftp)
+
+                self.__process_paths(ftp)
+
+                while True:
+                    time.sleep(.01)
+                    if self.__stopped:
+                        break
+
+        except Exception as e:
+            self.__log.exception(e)
+            try:
+                self.close()
+            except Exception as e:
+                self.__log.exception(e)
+        while True:
+            if self.__stopped:
+                break
+
+    def __connect(self, ftp):
+        try:
+            ftp.connect(self.host, self.port)
+
+            if isinstance(ftp, FTP_TLS):
+                ftp.sendcmd('USER ' + self.security[0])
+                ftp.sendcmd('PASS ' + self.security[1])
+                ftp.prot_p()
+                self.__log.info('Data protection level set to "private"')
+            else:
+                ftp.login(self.security[0], self.security[1])
+
+        except Exception as e:
+            self.__log.error(e)
+            time.sleep(10)
+        else:
+            self._connected = True
+            self.__log.info('FTP connected')
+
+    def __process_paths(self, ftp):
+        # TODO: call path on timer
+        for path in self.paths:
+            configuration = {
```

Comment:
Could we overwrite __dict__ method in Path class to get this data as a dictionary from Path instance?